CIPs/0016.md

```diff
@@ -0,0 +1,84 @@
+- enable simple construction of QRs by code from arbitrary Celo transactions
+
+## Proposed Solution
+
+### Syntax
+Celo protocol URLs contain "celo" in their schema (protocol) part and are constructed as follows:
+
+    request                 = "celo://" target [ "/" endpoint ] [ "?" parameters ]
+    
+    target                  = E164_PHONE_NUMBER / registry_id / celo_address
+    registry_id             = "StableToken" / "GoldToken" / "LockedGold" / ...
+    celo_address            = ( "0x" 40*HEXDIG )
+    
+    endpoint                = function_name "(" [ function_args ] ")
+    function_name           = "cUSD" / "CELO" / STRING
+    function_args           = function_arg *( "," function_arg )
+    function_arg            = value [ ":" TYPE ]
+
+    parameters              = parameter *( "&" parameter )
+    parameter               = key "=" value
+    key                     = "value" / "gas" / "gasPrice" / "feeCurrency" / "gatewayFee" / "gatewayFeeRecipient" / "localValue" / "localCurrencyCode" / "from" / "nonce"
+    value                   = NUMBER / STRING / celo_address
+
+### Semantics
+
+`TYPE` is a standard ABI type name, as defined in [Ethereum Contract ABI specification](https://solidity.readthedocs.io/en/develop/abi-spec.html). `STRING` is a URL-encoded unicode string of arbitrary length, where delimiters and the percentage symbol (`%`) are mandatorily hex-encoded with a `%` prefix.
+
+`E164_PHONE_NUMBER` is a standard E164 universal phone number, as defined by the International Telecommunication Union [E164 Number Specification](https://www.itu.int/rec/T-REC-E.164).
+
+`registry_id` is provided to simplify and shorten Celo protocol-aware contract interactions (to be eventually translated by the application to a registry-specified 40 byte address). The canonical list of registry identifiers can be obtained from the contract deployed at `0x000000000000000000000000000000000000ce10`. 
+
+`cUSD` and `CELO` are provided as a concise encoding of native ERC20 transfers that would otherwise be represented via a contract `registry_id`, `function_name`, and recipient `celo_address`. These optionally take 1 `function_arg` which is an amount in the token denomination.
+
+`function_args` have an optional `TYPE` because, in some cases, the application may have knowledge of the ABIs deeming the solidity types unnecessary. These can most likely be omitted when used with a Celo registry contract via a `registry_id` target.
+
+`localValue` is intended for use with `localCurrencyCode` and specifies a value in a locale-specific denomination (to be eventually translated by the application to a Celo-specific denomination). Resolving `value` with `localValue` is discretionary.
+
+All `key` values should be acommodated in abbreviations as the first character from each camelcase section. As an example, `value`, `localCurrency` and `gatewayFeeRecipient` could be provided as `v`, `lc` and `gfr`.  
+
```

Comment:
Hey @yorhodes ! 
From the gitcoin team here, just had a question regarding the format 

If I wanted to send :
**from**: 123
**to**: 456
**amount**: 10 cUSD

The qr string would look like : `celo://456/cUSD?v=10`
Would I be right in that ?

Also I see `from` as a parameter. I assume this is a from celo address but not super clear on how would happen if someone who isn't the owner scans the code